src/decrypt.rs

```diff
@@ -60,53 +58,38 @@ impl Decryptor {
     ///           - \<filename>:fd=\<filedescriptor> \
     ///           - \<filename>:\<password> \
     ///           - provider:<cmd/gprc>
-    pub async fn get_plaintext_layer(
+    pub fn get_plaintext_layer(
         &self,
         descriptor: &OciDescriptor,
         encrypted_layer: Vec<u8>,
         decrypt_config: &str,
-    ) -> Result<Vec<u8>> {
+    ) -> Result<Box<dyn Read + 'static>> {
         if !self.is_encrypted() {
-            return Err(anyhow!("unencrypted media type: {}", self.media_type));
-        }
-
-        if decrypt_config.is_empty() {
-            return Err(anyhow!("decrypt_config is empty"));
-        }
-
-        let cc = create_decrypt_config(vec![decrypt_config.to_string()], vec![])?;
-        let descript = descriptor.clone();
-
-        // ocicrypt-rs keyprovider module will create a new runtime to talk with
-        // attestation agent, to avoid startup a runtime within a runtime, we
-        // spawn a new thread here.
-        let handler = tokio::task::spawn_blocking(move || {
-            decrypt_layer_data(&encrypted_layer, &descript, &cc)
-        });
```

Comment:
This is still needed, since the `unwrap_key` in `ocicrypt-rs` keyprovider already created a new runtime, else it will have below error:
```
thread 'tests::test_image_rs' panicked at 'Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.0/src/runtime/enter.rs:39:9
```

The root cause is unwrap key operation for keyprovider is async, we may need seperate it as a pubic API to avoid block the whole decrypt processing.